doc/contributing.md

```diff
@@ -345,7 +345,7 @@ To avoid accidental highlighting follow this rule in ``*.nim`` files:
           The first 2 lines are for other RST implementations,
           including Github one.
 
-          .. [*] this is fulfilled when ``doc/rstcommon.rst`` is included.
+          \[*] this is fulfilled when ``doc/rstcommon.rst`` is included.
```

Comment:
Why are these backslashes needed?

Reply:
it's an RST thing that if you declare Rst footnote with symbols then you must reference it, leaving it as it is:
```
Error: number of footnotes and their references don't match: 1
```
Current PR breaks this feature because we cannot reference footnotes anymore — I could have added an exception for RST syntax, but it's futile considering that Markdown will have a different syntax for them anyway.
So I effectively simplified full rendering of footnotes into simple plaintext "footnotes" in 3 files where they were used
```
doc/contributing.md +348
rst.nim
docgen.md
```
until Markdown syntax is added.